discord/gateway.py

```diff
@@ -866,16 +866,23 @@ async def identify(self) -> None:
         await self.send_as_json(payload)
 
     @classmethod
-    async def from_client(
-        cls, client: VoiceClient, *, resume: bool = False, hook: Optional[Callable[..., Coroutine[Any, Any, Any]]] = None
+    async def from_connection_state(
+        cls,
+        state: VoiceConnectionState,
+        *,
+        resume: bool = False,
+        hook: Optional[Callable[..., Coroutine[Any, Any, Any]]] = None,
     ) -> Self:
         """Creates a voice websocket for the :class:`VoiceClient`."""
-        gateway = 'wss://' + client.endpoint + '/?v=4'
+        # TODO: Do I add a `if state.endpoint is None: raise Something` check here?
+        #       Putting None in there would be awkward, if that's even possible
```

Comment:
Doesn't seem necessary since it's being overwritten anyway

Reply:
Specifically I want to avoid the scenario where `None` somehow makes its way into that fstring but I guess it would error regardless at some point if the state was that bad.